src/main/java/com/thealgorithms/others/SieveOfEratosthenes.java

```diff
@@ -1,82 +1,48 @@
-package com.thealgorithms.others;
-
+import java.util.ArrayList;
 import java.util.Arrays;
 
-/**
- * Sieve of Eratosthenes is an ancient algorithm for finding all prime numbers
- * up to any given limit. It does so by iteratively marking as composite (i.e.,
- * not prime) the multiples of each prime, starting with the first prime number,
- * 2. The multiples of a given prime are generated as a sequence of numbers
- * starting from that prime, with constant difference between them that is equal
- * to that prime. This is the sieve's key distinction from using trial division
- * to sequentially test each candidate number for divisibility by each prime.
- * Once all the multiples of each discovered prime have been marked as
- * composites, the remaining unmarked numbers are primes.
- * <p>
- * Poetry about Sieve of Eratosthenes:
- * <p>
- * <i>Sift the Two's and Sift the Three's:</i></p>
- * <p>
- * <i>The Sieve of Eratosthenes.</i></p>
- * <p>
- * <i>When the multiples sublime,</i></p>
- * <p>
- * <i>The numbers that remain are Prime.</i></p>
- *
- * @see <a href="https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes">Wiki</a>
- */
 public class SieveOfEratosthenes {
 
     /**
      * @param n The number till which we have to check for prime Prints all the
      * prime numbers till n. Should be more than 1.
-     * @return array of all prime numbers between 0 to n
+     * @return ArrayList of all prime numbers between 0 to n
      */
-    public static int[] findPrimesTill(int n) {
-        // Create array where index is number and value is flag - is that number a prime or not.
-        // size of array is n + 1 cause in Java array indexes starts with 0
-        Type[] numbers = new Type[n + 1];
+    public static ArrayList<Integer> findPrimesTill(int n) {
```

Comment:
What if n is negative?

Reply:
If n is negative, the current implementation would throw an ArrayIndexOutOfBoundsException when creating the boolean array. To handle this case, we can add a check at the beginning of the method to throw an IllegalArgumentException with a message explaining that the limit must be at least 2.

Reply:
Yes please add validation